presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/thrift/InMemoryHiveMetastore.java

```diff
@@ -227,9 +227,17 @@ public synchronized void dropTable(MetastoreContext metastoreContext, String dat
         if (deleteData && table.getTableType().equals(MANAGED_TABLE.name())) {
             for (String location : locations) {
                 if (location != null) {
-                    File directory = new File(new Path(location).toUri());
-                    checkArgument(isAncestor(directory, baseDirectory), format("Table directory %s must be inside of the metastore base directory %s", directory, baseDirectory));
-                    deleteDirectory(directory);
+                    try {
+                        File directory = new File(new Path(location).toUri());
+                        checkArgument(isAncestor(directory, baseDirectory), format("Table directory %s must be inside of the metastore base directory %s", directory, baseDirectory));
+                        deleteDirectory(directory);
+                    }
+                    catch (IllegalArgumentException ex) {
```

Comment:
this seems a bit funny to throw an IllegalArgumentException and immediately catch it and change the message. Why not throw the desired exception in the first place?

Reply:
I agree. This is ugly. The specific IllegalArgumentException that can be thrown here (and was in the diff I was working on) comes from inside the JDK's File.toURI method. Since I can't plausibly change that exception's message, I did the next best thing and created a new exception of the same type with a new message.

Reply:
can you add a comment with this info?